Правда или действие

Lightdeed opened the pull request:
Первая итерация, прошу принять на предварительное review.

---
:mortar_board: [Правда или действие](https://up.htmlacademy.ru/javascript/30/user/2372159/tasks/16)

:boom: https://htmlacademy-javascript.github.io/2372159-kekstagram-30/11/

Commit:
1. Большинство переменных вынес в начало модуля,
как передать DOM со smallPictures до отработки функций пока не знаю.
smallPictures не вынес

2. Переименовал кнопки

3. В переменных с названием класс пока не нашел ничего лучше, как
переименовать Class в Element

4. вынес generatePhototsArray в модуль photosArray

Commit:
bu

Commit:
bu

Commit:
bu

Commit:
переименование

Commit:
bu

Commit:
Первая итерация, module9

Commit:
effectLevelElement' is assigned a value but never - fixed

Aleksandr-Bondarev commented:
1. Согласно тз: "При переключении эффектов, уровень насыщенности сбрасывается до начального значения (100%): слайдер, CSS-стиль изображения и значение поля должны обновляться."

При переключении на эффекты Марвин, Фобос и Зной значение применения эффекта и полоэение слайдера не 100%. Оригинал, Хром и Сепия - всё хорошо.

2. Благодаря информации об ошибке пользователю должно быть проще понять, какие критерии могли быть нарушены. В данно случае текст ошибки "НАРУШЕН ОДИН ИЗ ПУНКТОВ: 1, 2, 3, 4, 6" не поможет понять, что нужно исправить. Тексты ошибок можно составить из комбинаций требований, перечисленных в пунктах 2.3 и 2.4. https://up.htmlacademy.ru/javascript/30/project/kekstagram#kekstagram-2-3

3. Если попробовать загрузить картинку, применить к ней любой эффект (например, сепия), закрыть форму и попробовать загрузить тоже самое(!) изображение, применится фильтр сепия, но будет выбран "оригинал". В отправленной форме, в таком случае, так же будет отсутствовать фильтр.

4. Тернарный оператор лучше использовать для возврата значения. Конструкцию:
  isValid ? submitFormButton.disabled = false : submitFormButton.disabled = true;
 лучше заменить на условия.
 
5. При загрузке изображения, сначала появляется кот (тестовое фото) и только потом наше изображение. Лучше сначала добавлять подгружаемое фото в src, затем убирать класс .hidden.

6. Стоит убрать из кода закомментированные логи, экспорты, код, описания заданий и тд. Подобное присутствует в большинстве модулей. Комментарий уместен, если он помогает человеку, читающему код, понять происходящее, сообщает важную информацию о программе. В остальных случаях комментарии будут излишними.

Commit:
back up

Commit:
Учтено замечание 1:

При переключении на эффекты Марвин, Фобос и Зной значение применения
эффекта и положение ползунка слайдера не 100%.

Commit:
2. Дополнены тексты сообщений об ошибках

Commit:
добавлено обновление фильтра при каждой инициализации слайдера

Commit:
bu

Commit:
4. Тернарный оператор isValid ? submitFormButton.disabled = false
заменен на условия

Commit:
Пункт 6:
Лучше сначала добавлять подгружаемое фото в src, затем убирать класс .hidden.

Lightdeed commented:
Добавил костылей по пунктам 1-5.

Aleksandr-Bondarev commented:
1. В данный момент при переключении с фильтра на оригинал не сбрасываются стили, применённые к фото.
2. Иногда кот всё ещё появляется раньше, чем загружнное фото. Это может быть связано с тем, что фото подгружается не сразу после добавления ссылки в src. Чтобы было совсем хорошо, можно добавить слушатель события 'load' на imgUploadPreview, в котором мы будем вызывать функцию handleImageUpload. Тогда её вызов будет происходить только после завершения загрузки фото.

Также важно обратить внимание, что при каждой загрузке мы будем добавлять слушатель события 'load'. Чтобы слушатель события удалялся после первого срабатывания, можно передать в качестве тертьего параметра в addEventListener объект: { once: true }.

3. maxHashtagLength, hashtagsMaxCount, photoEffects, commentsLimit, namesArray, messagesArrayBasic - Названия констант (постоянных значений, известных до начала выполнения программы) написаны прописными (заглавными) буквами (критерий Б7).

Критерий Б10: Все константы выносятся в начало модуля. Перечисления, как набор констант, также выносятся. Это касается maxHashtagLength, hashtagsMaxCount, photoEffects.

4. В проекте два модуля common_functions.js и functions.js. В каждом из них находятся маленькие вспомогательные функции: получить случайное число, проверка на палиндром, отфильтровать числа и тд. Думаю, лучше объединить их в один модуль. Название common_functions.js мне кажется более информативным. Или, если функции из functions.js не будут использоваться в проекте, просто удалить лишний модуль.

Commit:
Пункт 4: удален файл неиспользуемых функций functions.js

Commit:
Учет 4х замечаний от 20231111 завершен

Commit:
bu

Aleksandr-Bondarev commented:
1. Из текста ошибок валидации можно убрать упоминание "нарушен пункт <номер>". Пользователю нужна только информация о том, что пошло не так.

2. Сейчас при нажатии на "+" и "-" у нас нет шага в 25%, фотография становится либо сразу максимально маленькой (25%), либо максимально большой (100%). Шаги 75% и 50% отсутствуют. https://up.htmlacademy.ru/javascript/30/project/kekstagram#kekstagram-2-1

3. Загружаем изображение, устанавливаем размер 25%, закрываем форму. Снова загружаем в инпут изображение,  ему устанавливаются стили как для 25%, при этом форма говорит, что установлен размер 100%. Нужно проверять сброс стилей при закрытии формы.

4. Константы должны быть первыми в модуле. Например:
- в модуле comments.js COMMENTS_ID_LIMIT объявлена после commentsIdsArray, должно быть наоборот;
- в модуле imageUploadFormEffects.js PHOTO_EFFECTS объявлена после dom-элементов;
- в модуле imageUploadFormScale SCALE_STEP объявлена после dom-элементов;
- в модуле imageUploadFormValidation то же самое с MAX_HASHTAG_LENGTH и HASHTAGS_MAX_COUNT;

Commit:
учет замечаний

Lightdeed commented:
Изменения выложил
